packages/core/mesh/teleport/src/muxing/muxer.ts

```diff
@@ -115,14 +116,25 @@ export class Muxer {
     };
 
     // NOTE: Make sure channel.push is set before sending the command.
-    this._sendCommand({
-      openChannel: {
-        id: channel.id,
-        tag: channel.tag,
-        contentType: channel.contentType,
+    this._sendCommand(
```

Comment:
why dont you just use promises?

Reply:
i've started with callbacks, since this is a default way of controlling streams, and kinda in process of converting to promises. i think i hit some issues and it just did not work quickly, so i left callback here for a moment